### PR TITLE
CC-14522: resolve hadoop, jackson, thrift, log4j, commons-collection, bean-utils, netty, zookeeper, commons-codec, and httpclient CVEs

### PR DESCRIPTION
**Pull request by levzem**
Signed-off-by: Lev Zemlyanov <lev@confluent.io>

## Problem
lots of CVEs leftover in 5.2.x and on

## Solution
resolve them using various methods
https://twistlock.tools.confluent-internal.io/#!/monitor/vulnerabilities/images/ci?search=storage-common:PR-185
all CVEs are resolved, the ones left are in S3 and HDFS
except for guava which i was unable to resolve because there is a backwards incompatible change that HDFS relies on
some CVEs have no resolution i.e `commons-collection`

##### Does this solution apply anywhere else?
- [x] yes
- [ ] no

##### If yes, where?
other CP connectors (S3 + HDFS)

## Test Strategy
tested against HDFS connector locally

##### Testing done:
- [x] Unit tests
- [x] Integration tests
- [ ] System tests
- [ ] Manual tests

## Release Plan

 
backport to `5.2.x`

### CODE REVIEW

**Comment by kkonstantine** [comment]
Last scan still shows a number of CVE unresolved. jackson databind 2.4.0 in the shaded jars as well as jackson asl and a few more. Have we confirmed that the fixes are ported accurately from recent CP versions?

**Comment by levzem** [comment]
@kkonstantine 

as mentioned above all the CVEs that are left are being brought in by S3 and HDFS, for some reason they are being displayed on `storage-common`, see below for an example

Screen Shot 2021-05-13 at 10 34 09 AM

a few other CVEs we are unable to fix such `asl`, 'guava', and `commons-collections` as no fix versions exist and are required by the connector to function.

File: format/pom.xml [hunk 1]
```diff
@@ -45,11 +51,59 @@
...
+                            <minimizeJar>false</minimizeJar>
+                            <createSourcesJar>false</createSourcesJar>
+                            <artifactSet>
+                                <includes>
+                                    <include>${jackson.groupId}:*</include>
+                                </includes>
+                            </artifactSet>
+                            <filters>
+                                <filter>
+                                    <artifact>${jackson.groupId}:*</artifact>
```
Comment (ilanjiR) [inline format/pom.xml:91]: Why are we filtering `${jackson.groupId}:*` if we're explicitly adding it above ^ ?

File: pom.xml [hunk 2]
```diff
@@ -60,21 +60,34 @@
...
+        <hadoop.version>2.10.1</hadoop.version>
         <hive.version>1.2.2</hive.version>
-        <joda.version>2.9.6</joda.version>
-        <licenses.version>5.2.5-SNAPSHOT</licenses.version>
-        <parquet.version>1.8.2</parquet.version>
         <httpclient.version>4.5.13</httpclient.version>
         <httpcore.version>4.4.4</httpcore.version>
+        <jackson.version>2.10.5</jackson.version>
+        <jackson.databind.version>2.10.5.1</jackson.databind.version>
+        <jackson.mapper.asl.version>1.9.13</jackson.mapper.asl.version>
```
Comment (ilanjiR) [inline pom.xml:81]: I'm guessing there's no universal version for these 3 modules that addresses all the CVEs

**Review by ilanjiR — COMMENTED** [review]
Couple of questions, but looks good otherwise

File: pom.xml (hunk 2 above)
Comment (levzem) [reply pom.xml:81]: correct

File: format/pom.xml (hunk 1 above)
Comment (kkonstantine) [reply format/pom.xml:91]: We need to first remove the old version shaded in the package and then reshade it with the version we supply here.

**Review by ilanjiR — APPROVED** [review]
LGTM

**Pull request merged**